File: frontend/src/lib/shoppingApi.js
import { RequestJson } from "./apiClient.js";

const ShoppingBase = "/shopping/items";

export const FetchShoppingItems = (householdId) =>
  RequestJson(`${ShoppingBase}?household_id=${encodeURIComponent(householdId)}`);

export const CreateShoppingItem = (payload) =>
  RequestJson(ShoppingBase, {
    method: "POST",
    body: JSON.stringify(payload)
  });

export const UpdateShoppingItem = (itemId, payload, householdId) =>
  RequestJson(`${ShoppingBase}/${itemId}?household_id=${encodeURIComponent(householdId)}`, {
    method: "PUT",
    body: JSON.stringify(payload)
  });

export const DeleteShoppingItem = (itemId, householdId) =>
  RequestJson(`${ShoppingBase}/${itemId}?household_id=${encodeURIComponent(householdId)}`, {
    method: "DELETE"
  });

File: apps/ios/EverdayIOS/EverdayIOS/Services/ShoppingApi.swift
import Foundation

enum ShoppingApi {
    static func fetchItems(householdId: Int, includeInactive: Bool = false) async throws -> [ShoppingItem] {
        let inactiveValue = includeInactive ? "true" : "false"
        return try await ApiClient.shared.request(
            path: "shopping/items?household_id=\(householdId)&include_inactive=\(inactiveValue)",
            requiresAuth: true
        )
    }

    static func createItem(_ request: ShoppingItemCreate) async throws -> ShoppingItem {
        try await ApiClient.shared.request(
            path: "shopping/items",
            method: "POST",
            body: request,
            requiresAuth: true
        )
    }

    static func updateItem(itemId: Int, householdId: Int, request: ShoppingItemUpdate) async throws -> ShoppingItem {
        try await ApiClient.shared.request(
            path: "shopping/items/\(itemId)?household_id=\(householdId)",
            method: "PUT",
            body: request,
            requiresAuth: true
        )
    }

    static func deleteItem(itemId: Int, householdId: Int) async throws {
        try await ApiClient.shared.requestVoid(
            path: "shopping/items/\(itemId)?household_id=\(householdId)",
            method: "DELETE",
            requiresAuth: true
        )
    }
}
